ONSdigital/dp-cli
# Update dp generation tool

### What

CI templates updated to pull golangci golangci-lint image and update Makefiles to run the linters when lint target is executed

### How to review

- Run dp generation tool to see that no changes take place against dp-hello-world:
  - Jump onto `feature/hello-world-library`, see https://github.com/ONSdigital/dp-hello-world/pull/18 on your local clone of dp-hello-world
  - in dp-cli
    - run `make build`
    - run generation tool against each app/library type:
      - dp-hello-world-controller for frontend controller app repo
      - dp-hello-world-api for api repo
      - dp-hello-world-event for event driven app repo
      - dp-hello-world-library for library repo
```
./dp generate-project --create-repository n --name "dp-hello-world-controller" --description "Example controller project generated by dp-cli" --go-version 1.19.6 --port 8124 --project-location ../dp-hello-world --type controller

./dp generate-project --create-repository n --name "dp-hello-world-api" --description "Example api project generated by dp-cli" --go-version 1.19.6 --port 8123 --project-location ../dp-hello-world --type api
     
./dp generate-project --create-repository n --name "dp-hello-world-event" --description "Example event driven app project generated by dp-cli" --go-version 1.19.6 --port 8123 --project-location ../dp-hello-world --type event-driven

./dp generate-project --create-repository n --name "dp-hello-world-library" --description "Example library project generated by dp-cli" --go-version 1.19.6 --port 8123 --project-location ../dp-hello-world --type library
```

- Lastly check for changes in your local dp-hello-world, run: `git diff` (any go.mod/go.sum changes could be because of any recent updates since creating PRs)

### Who can review

Anyone
